wgengine/magicsock/magicsock.go

```diff
@@ -3291,6 +3313,46 @@ func (c *Conn) UpdateStatus(sb *ipnstate.StatusBuilder) {
 	})
 }
 
+// updateStats updates the statistics counters with the src, dst, and cnts.
+// It is a noop if statistics gathering is not enabled.
+func (c *Conn) updateStats(src netip.Addr, dst netip.AddrPort, cnts tunstats.Counts) {
+	// This method is written to be inlineable.
```

Comment:
Will add tests if we think this is a feature we should provide. Might be a performance benchmark rather than a "go build" test (which are annoying to write).

Reply:
So is this kinda an abuse of the flowtrack.Tuple type? It might have the right shape, but the wrong names/docs?

Reply:
A test for this seems like a non-trivial amount of work. I just manually inlined this check in the 4 places that care.